hangman.rb

```diff
@@ -0,0 +1,133 @@
+require 'faker' #comment ln:50 && uncomment ln:51 for harry potter theme
+require 'random_word_generator'
+require 'colorize'
+
+# main class which initiates games and calls other classes based on user_response
+class GamePlay
+  attr_accessor :play, :wrong_guesses, :progress
+
+  def initialize(play)
+    @play = play
+  end
+
+  def quit_or_play_game
+    if @play == "n"
+      puts "\nGoodbye!"
+      exit
+    elsif @play == "y"
+      new_word = GenerateWord.new
+      word_array = new_word.create_fake_word
+      user_guess = Guess.new(word_array)
+      picture = Display.new(6)
+      picture.print_flower
+
+      while @play == "y"
+        print "Word: #{user_guess.progress}\n"
+        puts "Incorrect guesses: #{user_guess.wrong_guesses.join(" ")}"
+
+        if word_array.join(" ") == user_guess.progress
+            puts "\nCongrats! You win!"
+            exit
+          end
+
+        print "Enter letter > "
+        letter = gets.chomp.downcase
+        user_guess.compare(letter)
+      end
+    end
+  end
+end
+
+# creates a random word for the user to guess
+class GenerateWord
+  attr_accessor :word
+
+  def initialize
+    @word = word
+  end
+
+  def create_fake_word
+    @word = RandomWordGenerator.word
+    #@word = Faker::HarryPotter.character
+    @word = @word.split("")
+    return @word
+  end
+
+end
+
+class Guess
+  attr_accessor :letter, :word, :progress, :wrong_guesses
+
+  def initialize(word)
+    @word = word
+    progress = []
+    @wrong_guesses = []
+    @word.length.times do
+      progress.push("_")
+    end
+    @progress = progress.join(" ")
+  end
+
+  def compare(letter)
+    @letter = letter
+    if @letter == "quit"
+      quit_game = GamePlay.new("n")
+      quit_game.quit_or_play_game
+    elsif @word.include?(@letter)
+      @progress = progress.split(" ")
```

Comment:
Clever use of spaces (" ") to separate the underscores and removing them before you search for the letter.  This is a bit inefficient, because each deletion and insertion requires large portions of the array to be updated, but for small words it's reasonable.